feat: add OPENAPI_SPECTRAL_CONFIG_FILE to super-linter workflow for repo-specific Spectral rules

## Problem

Super-linter's OPENAPI linter runs Spectral with the built-in `spectral:oas` defaults, ignoring repo-specific `.spectral.yaml` configurations. This causes false positives when a repo has legitimate rule overrides.

## Evidence

`f5xc-salesdemos/api-specs` has a `.spectral.yaml` that:
- Disables the broken built-in `path-params` rule (Spectral issue #2767)
- Replaces it with a custom `f5-path-params` function that correctly handles F5's dotted parameter names like `{metadata.namespace}`
- Configures severity overrides for auto-fixable vs non-fixable rules

Despite `.spectral.yaml` existing at the repo root and `LINTER_RULES_PATH: "."` being set, the OPENAPI linter does not pick it up. It continues to use `spectral:oas` defaults and flags dotted parameter names as errors:

```
tests/fixtures/spectral_test_spec.json
 18:11  error  path-params  Parameter "metadata.namespace" must be used in path
```

This is the last remaining super-linter failure in `api-specs` — all other 21 linters pass.

## Proposed Fix

Add `OPENAPI_SPECTRAL_CONFIG_FILE` to the super-linter workflow environment variables, following the same pattern used for other linters:

```yaml
env:
  # Existing patterns:
  YAML_YAMLLINT_CONFIG_FILE: .yamllint.yaml
  PYTHON_RUFF_CONFIG_FILE: ruff.toml
  PYTHON_RUFF_FORMAT_CONFIG_FILE: ruff.toml
  PYTHON_PYLINT_CONFIG_FILE: .python-lint
  PYTHON_MYPY_CONFIG_FILE: .mypy.ini
  
  # Add:
  OPENAPI_SPECTRAL_CONFIG_FILE: .spectral.yaml
```

This tells super-linter to look for `.spectral.yaml` in the repo root (via `LINTER_RULES_PATH`) for the OPENAPI linter, just like it does for ruff, pylint, mypy, and yamllint.

## Why this matters

Without repo-specific Spectral config, any repo with F5 XC API specs will have false-positive `path-params` failures. The dotted parameter naming convention (`{metadata.namespace}`, `{metadata.name}`) is a fundamental F5 API design pattern used across all 268 domain specs. Disabling `path-params` and replacing it with a correct custom rule is the right fix — but it only works if super-linter reads the repo's `.spectral.yaml`.

## Governance note

`.spectral.yaml` is NOT currently in `governance.json` because each repo may have different Spectral rules. The env var just tells super-linter where to find the config — each repo controls its own content. This is the same model as `ruff.toml` (governed file) vs `pyproject.toml` (repo-specific overrides).

## Related

- f5xc-salesdemos/api-specs#99 — Spectral OAS3 integration with custom `f5-path-params` function
- f5xc-salesdemos/api-specs#109 — Super-linter compliance PR (21/22 linters pass, OPENAPI is the last one)
- Spectral issue stoplightio/spectral#2767 — built-in `path-params` bug with equivalent paths

--- a/.codespellrc
+++ b/.codespellrc
@@ -1,3 +1,3 @@
 [codespell]
-skip = *.json,*/tools/generated/*,vendor/*
-ignore-words-list = afterall,aks,datas,edn,hcl,ists,iterm,nd,preceeding,uncomplete,ves
+skip = *.json,*.lock,*/tools/generated/*,vendor/*
+ignore-words-list = afterall,aks,datas,edn,hcl,ists,iterm,nd,observ,pre-select,pre-selected,pre-selects,preceeding,uncomplete,ves

--- a/.github/workflows/super-linter.yml
+++ b/.github/workflows/super-linter.yml
@@ -79,3 +79,6 @@ jobs:
           PYTHON_RUFF_FORMAT_CONFIG_FILE: ruff.toml
           PYTHON_PYLINT_CONFIG_FILE: .python-lint
           PYTHON_MYPY_CONFIG_FILE: .mypy.ini
+          # ── OpenAPI / Spectral ──────────────────────────────
+          # Repos provide their own .spectral.yaml for custom rules.
+          OPENAPI_SPECTRAL_CONFIG_FILE: .spectral.yaml

--- a/.prettierignore
+++ b/.prettierignore
@@ -5,3 +5,9 @@
 **/*.gen.ts
 **/*.gen.js
 **/*.gen.d.ts
+
+# Build and release output
+dist/
+build/
+release/
+vendor/

